### PR TITLE
Use SwiftPM’s localization plan

### LINKED ISSUE #77: Use Swift Package Manager's support for localization
Instead of our custom solution: https://github.com/sindresorhus/Preferences/blob/main/Sources/Preferences/Localization.swift

See the initial attempt and feedback in https://github.com/sindresorhus/Preferences/pull/82

### PR DESCRIPTION
**Pull request by Kyle-Ye**
Close #77

Support for resources in packages depends on Swift 5.3 which we already met.

If we want to support old Xcode, we should probably use the old Strings File. But here I'm using String Catalog for simpleness which is only supported by Xcode 15 and means we should bump the Package.swift version to Swift 5.9 too.

The other way is not bump Swift version and use the old Strings File.

Here is the macOS CLI I use to generate Localizable.xcstrings.

```swift
struct Localization {
    ... 
    // The contents in this repo's Localization.swift file with some modification.
    // zh-CN -> zh-hans
    // zh-TW -> zh-hant
}

struct XCString: Encodable {
    let sourceLanguage = "en"
    let strings: Strings
    let version = "1.0"
}

struct Strings: Encodable {
    let preferences: LString
    let settings: LString
}

struct LString: Encodable {
    let extractionState = "manual"
    let localizations: [String: LanUnit]
}

struct LanUnit: Encodable {
    let stringUnit: StringUnit
}

struct StringUnit: Encodable {
    let state = "translated"
    let value: String
}

let encoder = JSONEncoder()

let value = XCString(strings: .init(
    preferences: .init(localizations: Localization.Identifier.preferences.value()),
    settings: .init(localizations: Localization.Identifier.settings.value())
))
let result = try encoder.encode(value)
let location = FileManager.default.homeDirectoryForCurrentUser.appending(path: "Downloads/Localizable.xcstrings")
try result.write(to: location)
print(location.absoluteString)

```

> Hint:
> Set CFBundleAllowMixedLocalizations to YES if the main app does not support multi-language
> See https://forums.swift.org/t/46685 and https://useyourloaf.com/blog/swift-package-string-localization/ for more info

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -3,20 +3,15 @@ import PackageDescription
...
+    defaultLocalization: "en",
+	platforms: [.macOS(.v10_13)],
 	products: [
-		.library(
-			name: "Settings",
-			targets: [
-				"Settings"
-			]
-		)
+		.library(name: "Settings", targets: ["Settings"])
```
Comment (sindresorhus) [inline Package.swift:9]: Don't make unrelated changes.

File: Sources/Settings/SettingsTabViewController.swift [hunk 2]
```diff
@@ -95,11 +95,10 @@ final class SettingsTabViewController: NSViewController, SettingsStyleController
 			} else {
 				let settings: String
 				if #available(macOS 13, *) {
-					settings = Localization[.settings]
+                    settings = NSLocalizedString("settings", bundle: .module, comment: "Settings")
```
Comment (sindresorhus) [inline Sources/Settings/SettingsTabViewController.swift:98]: Incorrect indentation.

File: Package.swift (hunk 1 above)
Comment (Kyle-Ye) [reply Package.swift:9]: Got it. I'll revert back it.

File: Sources/Settings/SettingsTabViewController.swift (hunk 2 above)
Comment (Kyle-Ye) [reply Sources/Settings/SettingsTabViewController.swift:98]: Got the diff. The original code use tab to indent while my new code use space to indent.

I'll change to use "tab" to follow the repo's general style.

image
